src/Voron/Impl/Journal/WriteAheadJournal.cs

```diff
@@ -581,6 +580,8 @@ public void ApplyLogsToDataFile(CancellationToken token, TimeSpan timeToWait)
 
                     long lastFlushedTransactionId = -1;
 
+                    // RavenDB-13302: we need to force a re-check this before we make decisions here
+                    _waj._env.ActiveTransactions.RecheckOldestTransaction();
```

Comment:
Why we don't need to call that before other usages of `ActiveTransactions.OldestTransaction` ?

Reply:
This happens after write transactions, and will do the fixup for everything, basically.
So we only need to have this in one location

Reply:
Can you elaborate on "This happens after write transactions"? Why is that the case?

Reply:
This is the flusher, and it is going to run after write transactions, so that will ensure that this gets update eventually.
The actual value we have here isn't really that important, the only thing that matter here is that we make some progress over time

Reply:
Okay, so this way we prevent from getting stuck. Makes sense. Thanks for clarification.

Reply:
Okay, so this way we prevent from getting stuck. Makes sense. Thanks for clarification.